File: src/exceptions.py
class ParserException(Exception):
    """Generic exception raised during the content parsing process."""


class InvalidTagParserException(ParserException):
    """Exception raised when an invalid parsing tag is encountered."""


class StartTokenNotFoundParserException(ParserException):
    """Exception raised when the start token is not found during parsing."""


class EndTokenNotFoundParserException(ParserException):
    """Exception raised when the end token is not found during parsing."""


class TagNotFoundParserException(ParserException):
    """Exception raised when a specific tag is not found during parsing."""


class TeamCatalogParserException(ParserException):
    """Exception raised during the parsing of the team catalog."""

File: src/exceptions/exceptions.js
class ParserException extends Error {
  constructor(message) {
    super(message);
    this.name = 'ParserException';
  }
}

class InvalidTagParserException extends ParserException {
  constructor(tag_name) {
    super(`Invalid parsing tag: '${tag_name}'.`);
    this.name = 'InvalidTagParserException';
  }
}

class StartTokenNotFoundParserException extends ParserException {
  constructor(tag_name) {
    super(`Start token not found for tag: '${tag_name}'.`);
    this.name = 'StartTokenNotFoundParserException';
  }
}

class EndTokenNotFoundParserException extends ParserException {
  constructor(tag_name) {
    super(`End token not found for tag: '${tag_name}'.`);
    this.name = 'EndTokenNotFoundParserException';
  }
}

class TagNotFoundParserException extends ParserException {
  constructor(tag_name) {
    super(`Tag '${tag_name}' not found during parsing.`);
    this.name = 'TagNotFoundParserException';
  }
}

class TeamCatalogParserException extends ParserException {
  constructor(message) {
    super(`Error during team catalog parsing: ${message}`);
    this.name = 'TeamCatalogParserException';
  }
}

module.exports = {
  ParserException,
  InvalidTagParserException,
  StartTokenNotFoundParserException,
  EndTokenNotFoundParserException,
  TagNotFoundParserException,
  TeamCatalogParserException
};
